Bottom: b668e1dc49999a3a4a3b89ec1bbbdbf6eeb42ea1
Top:    22d5c5f48710b88acfe1daa41a318a85cb952581
Author: Jason Wang <jasowang@redhat.com>
Date:   2017-03-22 19:43:15 +0800

Refresh of vhost_net-prefetch-desc

---

diff --git a/drivers/vhost/net.c b/drivers/vhost/net.c
index 64601a7..3047781 100644
--- a/drivers/vhost/net.c
+++ b/drivers/vhost/net.c
@@ -774,29 +774,22 @@ static void handle_rx_batched(struct vhost_net *net, struct vhost_log *vq_log)
 					    + vhost_hlen + sock_hlen), i);
 		}
 		for (i = 0; i < avails; i++) {
-			if (nvq->rh == nvq->rt) {
-				printk("wrong!\n");
-			}
 			head = vhost_get_vq_desc2(vq, vq->iov,
 						  ARRAY_SIZE(vq->iov),
 						  &out, &in, vq_log,
 						  &log, indices[i]);
-			if (unlikely(head < 0 || head == vq->num)) {
-				printk("head error!\n");
+			if (unlikely(head < 0 || head == vq->num))
 				return;
-			}
 			if (rx_recvmsg(nvq, in, nvq->rxq[nvq->rh++],
 				       lens[i], vq_log, log,
-				       vhost_hlen, sock_hlen)) {
-				printk("recvmsg error!\n");
+				       vhost_hlen, sock_hlen))
 				return;
-			}
 
 			vhost_update_used_idx(vq, 1);
-			vhost_signal(&net->dev, vq);
-
 			/* FIXME: count bytes */
 		}
+		/* FIXME: batched signal */
+		vhost_signal(&net->dev, vq);
 	}
 }
